test/services/user-service.test.js

```diff
@@ -0,0 +1,40 @@
+let userService
+let MockSequelize
+let mockDb
+const testEmail = { email: 'test@example.com' }
+
+describe('User Service', () => {
+  beforeAll(async () => {
+    jest.mock('../../server/models')
+    jest.mock('../../server/models/user', () => {
+      MockSequelize = require('sequelize-mock')
+      mockDb = new MockSequelize('', '', '', { })
+      return mockDb.define('users', {
+        userId: 1,
+        email: 'test@example.com'
+      })
+    })
+    userService = require('../../server/services/user-service')
+  })
+
+  beforeEach(async () => {
+  })
+
+  test('create should create new user if one does not exist', async () => {
+    mockDb.$queueResult([ ])
+    const result = await userService.create(testEmail)
+    // the response from userService is inconsistent. If a user is created a boolean true is returned
+    expect(result).not.toBeFalsy()
+  })
+
+  test('create should return existing user if one exists', async () => {
+    mockDb.$queueResult({
+      userId: 1,
+      email: 'test@example.com'
+    })
+    await userService.create(testEmail)
+    const result = await userService.create(testEmail)
+    // the response from userService is inconsistent. If a user exists, the raw sequelize result is returned
+    expect(result).toBeDefined()

```

Comment:
We don't seem to be testing that the correct query was sent to Sequelize so the tests could miss something like incorrect query parameters causing false positives when checking for an existing user.

As an example, I made the following change in `server/services/user-service.js` and all tests still pass:
```
 module.exports = {
   create: async function (user) {
-    const existingUser = await userRepository.getByEmail(user.email)
+    const existingUser = await userRepository.getByEmail()^M
     if (existingUser != null) {
       console.log('existing user, no action required')
       return existingUser
     }
     console.log('creating new user')
     return userRepository.create(user)
   }
 }
```